### ISSUE TITLE
Add configuration option to control error verbosity for Razor language server

### ISSUE THREAD

**Issue by NTaylorMullen** [dotnet/aspnetcore#23550]
We should look into having a Razor specific configuration section to help control the vairous aspects of the editor.

@ToddGrun @alexgav @jimmylewis this might be something you guys want to do too

**Comment by ryanbrandenburg** [comment]
How committed are we to Tools->Options in VS? It was suggested that we might instead want to control this with a config file. That has the benefit of being cross-IDE, though it's probably less discover-able than what might be people's first instinct.

**Comment by NTaylorMullen** [comment]
> That has the benefit of being cross-IDE, though it's probably less discover-able than what might be people's first instinct.

So if this was just for Razor language server verbosity I'd say a config file would more than fine. However, given we're going to want to add Razor configurations for tabs/spaces, formatting etc. I don't want to commit to just a config file if we're going to have different options in the future.

Now that being said, if we can make VS options inferrable from a workspaces configuration options then that'd mean we could start out with a config and surface their settability in the UI which would mean using a config approach at first would be fine.

**Comment by ryanbrandenburg** [comment]
Talked to Taylor offline. Once we realized the scope of adding an options page (UI review, accessibility, localization, all in addition to the actual coding itself) we decided that it's not yet worth the investment. Considering that the usage scenario here is almost certainly that a dev has run into a problem and wants to take a look at their logs (or have us look at them) we think that it should be fine to have an Environment variable drive this for now. In the future we'll start exposing things like formatting options and at that point logging can jump on to that effort.
